eea/src/main/java/org/web3j/protocol/eea/crypto/PrivateTransactionEncoder.java

```diff
@@ -76,15 +72,23 @@
                 TransactionEncoder.asRlpValues(
                         privateTransaction.asRawTransaction(), signatureData));
 
-        result.add(RlpString.create(privateTransaction.getPrivateFrom()));
+        privateTransaction.getPrivateFrom().ifPresent(
```

Comment:
It may be better to have RawPrivateTransaction store the PrivateFrom/For as byte[32] (or something akin to 'PrivacyAddress' which wraps a 32-byte data-item), rather than as as String.

The Base64 decoding resolves prior problems regarding character encoding - however, a type-safe interface may offer benefits, rather than assuming the user of the API will know to provide a Base64 encoded string.

Reply:
Agreed

Reply:
done